crates/rpc-types-beacon/src/requests.rs

```diff
@@ -0,0 +1,150 @@
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "ssz", derive(ssz_derive::Decode, ssz_derive::Encode))]
+/// An Electra-compatible execution requests payload.
+pub struct ExecutionRequestsV4 {
+    /// The requested deposits.
+    pub deposits: Vec<DepositRequest>,
+    /// The requested withdrawals.
+    pub withdrawals: Vec<WithdrawalRequest>,
+    /// The requested consolidations.
+    pub consolidations: Vec<ConsolidationRequest>,
+}
+
+#[cfg(feature = "ssz_types")]
+mod from_requests {
+    use crate::requests::ExecutionRequestsV4;
+    use alloy_eips::{
+        eip6110::DepositRequest, eip7002::WithdrawalRequest, eip7251::ConsolidationRequest,
+        eip7685::Requests,
+    };
+    use ssz::DecodeError;
+
+    impl TryFrom<&Requests> for ExecutionRequestsV4 {
+        type Error = Error;
+        fn try_from(value: &Requests) -> Result<Self, Self::Error> {
+            use alloy_eips::{
+                eip6110::DEPOSIT_REQUEST_TYPE, eip7002::WITHDRAWAL_REQUEST_TYPE,
+                eip7251::CONSOLIDATION_REQUEST_TYPE,
+            };
+            use ssz::Decode;
+            use ssz_types::{typenum, VariableList};
+
+            type MaxConsolidationRequestsPerPayload = typenum::U1;
+            type MaxDepositRequestsPerPayload = typenum::U8192;
+            type MaxWithdrawalRequestsPerPayload = typenum::U16;
+
+            let (deposits, withdrawals, consolidations) = value.iter().try_fold(
+                (Vec::new(), Vec::new(), Vec::new()),
+                |mut acc, request| {
+                    if request.is_empty() {
+                        return Err(Error::EmptyRequest);
+                    }
+
+                    match request[0] {
+                        DEPOSIT_REQUEST_TYPE => {
+                            let list: VariableList<
+                                DepositRequest,
+                                MaxWithdrawalRequestsPerPayload,
+                            > = VariableList::from_ssz_bytes(&request[1..])?;
+                            acc.0.extend(Vec::<DepositRequest>::from(list));
+                        }
+                        WITHDRAWAL_REQUEST_TYPE => {
+                            let list: VariableList<
+                                WithdrawalRequest,
+                                MaxDepositRequestsPerPayload,
+                            > = VariableList::from_ssz_bytes(&request[1..])?;
+                            acc.1.extend(Vec::<WithdrawalRequest>::from(list));
+                        }
+                        CONSOLIDATION_REQUEST_TYPE => {
+                            let list: VariableList<
+                                ConsolidationRequest,
+                                MaxConsolidationRequestsPerPayload,
+                            > = VariableList::from_ssz_bytes(&request[1..])?;
+                            acc.2.extend(Vec::<ConsolidationRequest>::from(list));
```

Comment:
I believe there are ssz impls for `Vec<T>` so we shouldn't need ssz_types here?

Reply:
Oh perfect, I'll give that a shot, and then remove the `ssz_types` dep and merge the two conditional modules into one dependent on just `ssz`.

Reply:
Ok thanks @klkvr I was able to remove the ssz_types dep and just go with ssz.  I also added `const` for the payload limits to the appropriate EIP crates and moved `cfg_eval` to a workspace dep that's conditional on `serde` feature to clean up some warnings w/ `--no-default-features`.  PTAnotherL!